### PR TITLE
Allow customizing number of components in GrabCut algo

### PR DESCRIPTION
**Pull request by georgthegreat**
- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch

### CODE REVIEW

File: modules/imgproc/src/grabcut.cpp [hunk 1]
```diff
@@ -104,12 +111,51 @@ GMM::GMM( Mat& _model )
...
+    covDeterms = new double[componentsCount];
+    for( int ci = 0; ci < componentsCount; ci++ ) {
+        inverseCovs[ci] = new double*[3];
+        prods[ci] = new double*[3];
+        sums[ci] = new double[3];
+        for( int j = 0; j < 3; ++j ) {
+            inverseCovs[ci][j] = new double[3];
+            prods[ci][j] = new double[3];
+        }
+    }
```
Comment (asmorkalov) [inline modules/imgproc/src/grabcut.cpp:134]: It generates too many new calls with small fragments. I propose to use std::vector<cv::Matx33d>. All Matx objects are allocated "in place". We'll get single allocation and simple indexing. You have to replace [] with () in several cases to follow Matx operators semantic.

**Comment by georgthegreat** [comment]
NB: this PR is not lost, I will return to it in a while.

**Comment by asmorkalov** [comment]
@georgthegreat Friendly reminder.

**Comment by asmorkalov** [comment]
@georgthegreat Friendly reminder.

**Comment by georgthegreat** [comment]
Sorry, still overloaded with other tasks.
I do remember about the fix.

**Comment by asmorkalov** [comment]
@georgthegreat Friendly reminder.

**Comment by georgthegreat** [comment]
Let me go forward and close this.
I have no time to fix this, sorry about that.

**Pull request closed without merging**